H5P.BranchedVideo/branched-video.js

```diff
@@ -47,6 +47,69 @@ H5P.BranchedVideo = (function ($) {
       this.currentTime = 0.00;
       this.watchedTime = 0.00;
       this.nodes = [];  // array of node objects
+
+      this.playedTime = [[0,0]];
+
+      this.getPlayedTime = function(){
+        return this.playedTime;
+      }
+
+      this.updatePlayedTime = function(time){
+        var roundedTime = Math.round( time * 10 ) / 10; // make it 1 decimal
+        var arrayOfPlayedTime = this.playedTime;
+        var index = getIndex();
+        updateArray();
+        this.playedTime = arrayOfPlayedTime;
+
+        function getIndex(){
+          var length = arrayOfPlayedTime.length;
+          for (var i = 0 ; i<length; i++){
+            var bottom = arrayOfPlayedTime[i][0];
+            var top = arrayOfPlayedTime[i][1];
+            if (time <= top + 1 && time >= bottom){
+              return i;
+            }
+          }
+          return insertPair();
+        }
+
+        function insertPair(){
+          var length = arrayOfPlayedTime.length;
+          for (var j = 0; j<length-1; j++){
+            var currentTop = arrayOfPlayedTime[j][1];
+            var nextBottom = arrayOfPlayedTime[j +1][0];
+            if ( time > currentTop && time < nextBottom){
+              arrayOfPlayedTime.push([roundedTime, roundedTime + 0.1]);
+              for (var i = j+1; i < length+1; i++){
+                //sorts it
+                var temp = arrayOfPlayedTime[length];
+                arrayOfPlayedTime[length] = arrayOfPlayedTime[i];
+                arrayOfPlayedTime[i] = temp;
+              }
+              return j+1;
+            }
+          }
+          arrayOfPlayedTime.push([roundedTime,roundedTime +0.1]);
+          return length;
+        }
+
+        function updateArray(){
```

Comment:
The name `updateArray` is very generic. Can you please name it something more descriptive?